aptos-move/aptos-abstract-gas-usage/src/algebra_helpers.rs

```diff
@@ -79,46 +79,48 @@ pub fn expand_terms(expr: DynamicExpression) -> Vec<DynamicExpression> {
 ///
 /// (A + A) * (5 + 5) => 5A + 5A + 5A + 5A => 20A
 pub fn aggregate_terms(terms: Vec<DynamicExpression>) -> Result<BTreeMap<String, u64>> {
+    use DynamicExpression::*;
+
     let mut map: BTreeMap<String, u64> = BTreeMap::new();
     for term in terms {
         match term {
-            DynamicExpression::GasValue { value } => {
-                // seeing a concrete quantity means that the user
-                // isn't providing expressions in the GasExpression.
-                // this makes calibration impossible and we should error
+            GasValue { value } => {
+                // Seeing a concrete quantity means that the user is not encoding the gas
+                // cost in an abstract way using GasExpression.
+                //
+                // This makes calibration impossible and we should error.
                 if value != 0 {
-                    return Err(anyhow!(
-                        "Concrete quantity provided. Should be GasExpression."
-                    ));
+                    bail!(
+                        "Expected simple term got non-zero concrete value {}.",
```

Comment:
Do we all know what "simple term" means?  I assume it's (constant * var) ?

Reply:
Agreed, we should probably define "simple term" for clarity

Reply:
Revised the comments and now they clearly explain what a simple term means.